draft-ietf-quic-transport.md

```diff
@@ -1414,50 +1414,47 @@ Handshake[0]: CRYPTO[FIN], ACK[0]
+field to specify the connection ID that is used in the DCID field of packets
+being sent to them. Upon receiving a packet, each endpoint sets the DCID it
+sends to match the value of the SCID that it receives.
 
 When an Initial packet is sent by a client that has not previously received an
-Initial or Retry packet from the server, it populates the Destination Connection
-ID field with an unpredictable value.  This MUST be at least 8 bytes in length.
-Until a packet is received from the server, the client MUST use the same value
-unless it abandons the connection attempt and starts a new one. The initial
-Destination Connection ID is used to determine packet protection keys for
-Initial packets.
+Initial or Retry packet from the server, the client populates the DCID field
+with an unpredictable value.  This DCID MUST be at least 8 bytes in length.
+Until a packet is received from the server, the client MUST use the same DCID
+value on all packets in this connection. This DCID is used to determine packet
+protection keys for Initial packets.
 
-The client populates the Source Connection ID field with a value of its choosing
-and sets the SCID Len field to indicate the length.
+The client populates the SCID field with a value of its choosing and sets the
+SCID Len field to indicate the length.
 
-The first flight of 0-RTT packets use the same Destination and Source Connection
-ID values as the client's first Initial.
+The first flight of 0-RTT packets use the same DCID and SCID values as the
+client's first Initial packet.
 
 Upon first receiving an Initial or Retry packet from the server, the client uses
-the Source Connection ID supplied by the server as the Destination Connection ID
-for subsequent packets, including any subsequent 0-RTT packets.  That means that
-a client might change the Destination Connection ID twice during connection
-establishment, once in response to a Retry and once in response to the first
-Initial packet from the server. Once a client has received an Initial packet
-from the server, it MUST discard any packet it receives with a different Source
-Connection ID.
-
-A client MUST only change the value it sends in the Destination Connection ID in
-response to the first packet of each type it receives from the server (Retry or
-Initial); a server MUST set its value based on the Initial packet.  Any
-additional changes are not permitted; if subsequent packets of those types
-include a different Source Connection ID, they MUST be discarded.  This avoids
-problems that might arise from stateless processing of multiple Initial packets
-producing different connection IDs.
-
-The connection ID can change over the lifetime of a connection, especially in
-response to connection migration ({{migration}}); see {{issue-cid}} for details.
+the SCID supplied by the server as the DCID for subsequent packets, including
+all subsequent 0-RTT packets. This means that a client might change the DCID
```

Comment:
I think it might just be best to explicitly state which packet types require use of the new DCID. Not mentioning handshake packets here just leaves it unclear IMO.

Reply:
I agree, and as I noted in the description, this PR exposes this issue.

Ok, I'm going to give it a shot based on what I understand our intent was; I'll see how it goes.